core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -110,76 +110,76 @@ def start_board(self, boards: List[Tuple[FlightControllerType, str]]) -> bool:
-        if "endpoints" not in self.configuration:
-            self.configuration["endpoints"] = []
-        endpoints = self.configuration["endpoints"]
-        for endpoint in endpoints:
+        for endpoint in self._get_configuration_endpoints():
             try:
-                if not self.mavlink_manager.add_endpoint(Endpoint(**endpoint)):
+                if not self.mavlink_manager.add_endpoint(endpoint):
                     raise ValueError("Cannot add endpoint.")
             except Exception as error:
                 warn(f"Could not load endpoint {endpoint}. {error}")
 
-    def _reset_endpoints(self, endpoints: List[Endpoint]) -> None:
+    def _reset_endpoints(self, endpoints: Set[Endpoint]) -> None:
         try:
             self.mavlink_manager.clear_endpoints()
             self.mavlink_manager.add_endpoints(endpoints)
             print("Reseting endpoints to previous state.")
         except Exception as error:
             warn(f"Error reseting endpoints: {error}")
 
-    def _update_endpoints(self, updated_endpoints: List[Endpoint]) -> bool:
+    def _update_endpoints(self) -> bool:
         try:
-            self.configuration["endpoints"] = deepcopy(updated_endpoints)
+            self._save_endpoints_to_configuration(self.get_endpoints())
             self.settings.save(self.configuration)
             return self.restart()
         except Exception as error:
             warn(f"Error updating endpoints: {error}")
             return False
 
-    def get_endpoints(self) -> List[Endpoint]:
+    def get_endpoints(self) -> Set[Endpoint]:
         """Get all endpoints from the mavlink manager."""
         return self.mavlink_manager.endpoints()
 
-    def add_new_endpoints(self, new_endpoints: List[Endpoint]) -> bool:
+    def add_new_endpoints(self, new_endpoints: Set[Endpoint]) -> bool:
         """Add multiple endpoints to the mavlink manager and save them on the configuration file."""
-        saved_endpoints = deepcopy(self.configuration["endpoints"])
-        loaded_endpoints = deepcopy(self.get_endpoints())
+        loaded_endpoints = self.get_endpoints()
+
+        duplicated_endpoints = loaded_endpoints.intersection(new_endpoints)
+        if duplicated_endpoints:
+            warn(f"Endpoints {duplicated_endpoints} already loaded. Reseting operation.")
+            return False
 
         for endpoint in new_endpoints:
-            if endpoint.asdict() in saved_endpoints:
-                warn(f"Endpoint {endpoint} already stored.")
-                self._reset_endpoints(loaded_endpoints)
-                return False
             try:
                 self.mavlink_manager.add_endpoint(endpoint)
-                saved_endpoints.append(endpoint.asdict())
                 print(f"Adding endpoint {endpoint} and saving it to the settings file.")
             except NotImplementedError:
                 print(f"Failed to add endpoint {endpoint}. Connection_type not compatible with current router.")
```

Comment:
it would be nice to print the name of the router instead of "current router"

Reply:
Yes